sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -412,8 +414,8 @@ object Decimal {
     value match {
       case j: java.math.BigDecimal => apply(j)
       case d: BigDecimal => apply(d)
-      case k: scala.math.BigInt => apply(k)
-      case l: java.math.BigInteger => apply(l)
+      case k: BigInt => apply(k)
```

Comment:
I agree it's better to make it consistent, but let's not mix it into a simple fix for java 7 ...